src/controller.ts

```diff
@@ -211,8 +211,17 @@ export class Controller {
     function storeSign(node: Cell, sigCell: Cell): Cell {
       const slice = node.beginParse();
       if (slice.remainingRefs > 0) {
-        const child = storeSign(slice.loadRef(), msgToSign);
-        return slice.asBuilder().storeRef(child).asCell();
+        const child = storeSign(slice.loadRef(), sigCell);
+        const refs = [];
+
+        while (slice.remainingRefs > 0) {
+          refs.push(slice.loadRef());
+        }
+
+        const builder = slice.asBuilder().storeRef(child);
+        refs.forEach((ref) => builder.storeRef(ref));
+
+        return builder.asCell();
```

Comment:
- There aren't supposed to be any additional refs. instead, we can assert that there is only one ref.
- Instead of reconstructing, why not just push directly to refs once we found the deep most cell? (recursive or iterative, both can work)

Reply:
I think it'd be better to use the previous iterative method, if ton13 doesn't prevent it somehow

Reply:
In new `ton-core`, `refs` property of `Cell` is frozen and overall, it has been designed as a immutable object. So, in order to store signed message in the deepest cell, it seems inevitable to reconstruct the whole tree. 
For this scenario I think recursive approach is cleaner than an iterative one.